### PR TITLE
add md docs for 12 layers/criterions

### PR DESCRIPTION
**Pull request by shane-huang**
This replaces the previous pull request. https://github.com/intel-analytics/BigDL/pull/1076 

added 12 layers.

simple layers: Consine,MM,Mean,Sum
merge layers: CMaxTable
Pooling layers: RoiPooling,SpatialAveragePooling,
Loss functions: DiceCoefficientCriterion,MultiMarginCriterion,DistKLDivCriterion
Convolution layers: SpatialFullConvolution,SpatialConvolution

### CODE REVIEW

**Comment by shane-huang** [comment]
Update of the issues: 

Major issues:
the python examples of RoiPooling and SpatialFullConvolution needs to be fixed because of program throws error, issue #1066 #1063 needs to be fixed.

Minor issues:
SpaticalFullConvolution, SpatialConvolution python and scala API parameter list unsync. #1056
Mean, Sum missing "squeeze" argument in python API. #1029
DiceCoefficientCriterion default argument value is missing in python API #1029
Docs goes exactly the same as code (kept the difference if API's differ).

**Comment by shane-huang** [comment]
replaced this pull request with #1122

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+```scala
+
+scala>
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor._
+import com.intel.analytics.bigdl.tensor.Storage
+
+val m = SpatialFullConvolution[Tensor[Float],Float](1, 2, 2, 2, 1, 1,0, 0, 0, 0, 1, false)
```
Comment (psyyz10) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md]: This is layer has a bug, it should not use this kind of type declaration and I will open an issue.

File: docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+## DiceCoefficientCriterion ##
+
+**Scala:**
+```scala
+val loss = DiceCoefficientCriterion(sizeAverage=true, epsilon=1.0f)
+```
+**Python:**
+```python
+loss = DiceCoefficientCriterion(size_average,epsilon)
```
Comment (psyyz10) [inline docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md]: default value?

File: docs/docs/APIdocs/layers/nn/Pooling Layers/RoiPooling.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+print "input is :",[input_data, input_rois]
+
+m = RoiPooling(3,2,1.0)
+out = m.forward([input_data,input_rois])
+print "output of m is :",out
+```
+produces output:
+
+```python
+input is : [array([[[[ 0.08500103,  0.33421796,  0.29084699,  1.60344635, -0.24289341,
```
Comment (psyyz10) [inline docs/docs/APIdocs/layers/nn/Pooling Layers/RoiPooling.md]: Is the input a little huge?

File: docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md (hunk 2 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md]: This is a problem with DiceCoefficientCriterion - Scala has defaults while Python don't. Pls refer to "minor issues" at my comment. When the API changes the docs will be changed in sync. For now I just kept those difference in doc to make it sync with actual code.

File: docs/docs/APIdocs/layers/nn/Pooling Layers/RoiPooling.md (hunk 3 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Pooling Layers/RoiPooling.md]: RoiPooling needs to specify regions of interest in the original picture. So we need a larger input to better demonstrate the roi regions. I used the same size as in unit test.

File: docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md (hunk 2 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/DiceCoefficientCriterion.md]: see issue #1029

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md (hunk 1 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md]: I just merge the doc as is now, to keep in sync with code. Later when the layer is fixed, doc should be changed together in the same PR.
